proofs/README.md

```diff
@@ -0,0 +1,9 @@
+[//]: # (SPDX-License-Identifier: CC-BY-4.0)
+
+# Proofs for mldsa-native
+
+This directory contains material related to the formal verification of the source code of mldsa-native.
+
+## C verification: CBMC
+
+We use the [C Bounded Model Checker (CBMC)](https://github.com/diffblue/cbmc) to show the absence of various classes of undefined behaviour in the mldsa-native C source, including out of bounds memory accesses and integer overflows. See [proofs/cbmc](cbmc).
```

Comment:
Shall we reference the proof_guide written for mlkem-native here https://github.com/pq-code-package/mlkem-native/blob/main/proofs/cbmc/proof_guide.md?

Reply:
added in [19d6945](https://github.com/pq-code-package/mldsa-native/pull/385/commits/19d694555e1d7636a1e9254b2837256251e4b598)